galasa-parent/dev.galasa.framework.api.webui/src/main/java/dev/galasa/framework/api/webui/WorklistUtils.java

```diff
@@ -0,0 +1,214 @@
+package dev.galasa.framework.api.webui;
+
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import dev.galasa.framework.spi.DssAdd;
+import dev.galasa.framework.spi.DssSwap;
+import dev.galasa.framework.spi.DynamicStatusStoreException;
+import dev.galasa.framework.spi.FrameworkPropertyFileException;
+import dev.galasa.framework.spi.IDynamicStatusStoreService;
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.IResultArchiveStoreDirectoryService;
+import dev.galasa.framework.spi.IRunResult;
+import dev.galasa.framework.spi.ResultArchiveStoreException;
+
+public class WorklistUtils {
+	
+	private IFramework framework;
+	
+	public WorklistUtils(IFramework framework) {
+		this.framework = framework;
+	}
+	
+	
+	protected List<String> getRunIdsFromDss() throws DynamicStatusStoreException, ServletException {
+	
+		IDynamicStatusStoreService dss = framework.getDynamicStatusStoreService("webui");
+		String worklistProperty = dss.get("user.username.worklist");
+		
+		if (worklistProperty != null) {
+			
+			List<String> runIdsInWorklist = new ArrayList<String>();	
+			
+			if (worklistProperty.length() != 0) {
+				
+				JsonObject jsonObject = new JsonParser().parse(worklistProperty).getAsJsonObject();
+				
+				JsonArray runIds = (JsonArray) jsonObject.get("runIds");		
+
+				for (JsonElement el : runIds) {
+					runIdsInWorklist.add(el.getAsString());
+				}	
+			}	
+			
+			return runIdsInWorklist;
+			
+		} else {
+			return null;
+		}
+		
+	}
+	
+	
+	protected List<WorklistItem> getWorklistItems() throws ServletException {
+		
+		List<WorklistItem> runs = new ArrayList<WorklistItem>();
+		
+		if (WorklistQuery.runIdsInWorklist != null) {
```

Comment:
wooooo,   no way.     A servlet container is multithreaded,   so it will be completely random which "list" you will obtain from what thread.   Why not just pass the list as a parameter?